azurerm/internal/services/network/vpn_gateway_connection_resource.go

```diff
@@ -0,0 +1,643 @@
+								ValidateFunc: validate.VirtualHubRouteTableID,
+							},
+						},
+					},
+				},
+			},
+
+			"vpn_link_connection": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"vpn_site_link_id": {
+							Type:     schema.TypeString,
+							Required: true,
+							// The vpn site link associated with one link connection can not be updated
+							ForceNew:     true,
+							ValidateFunc: validate.VpnSiteLinkID,
+						},
+
+						"route_weight": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(0),
+							Default:      0,
+						},
+
+						"protocol": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.IKEv1),
+								string(network.IKEv2),
+							}, false),
+							Default: string(network.IKEv2),
+						},
+
+						"bandwidth_mbps": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(1),
+							Default:      10,
+						},
+
+						"shared_key": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"bgp_enabled": {
+							Type:     schema.TypeBool,
+							ForceNew: true,
+							Optional: true,
+							Default:  false,
+						},
+
+						"use_policy_based_traffic_selector": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"ipsec_policy": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"sa_lifetime_sec": {
+										Type:         schema.TypeInt,
+										Required:     true,
+										ValidateFunc: validation.IntBetween(300, 172799),
+									},
+									"sa_data_size_kb": {
+										Type:         schema.TypeInt,
+										Required:     true,
+										ValidateFunc: validation.IntBetween(1024, 2147483647),
+									},
+									"ipsec_encryption_algorithm": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(network.IpsecEncryptionAES128),
+											string(network.IpsecEncryptionAES192),
+											string(network.IpsecEncryptionAES256),
+											string(network.IpsecEncryptionDES),
+											string(network.IpsecEncryptionDES3),
+											string(network.IpsecEncryptionGCMAES128),
+											string(network.IpsecEncryptionGCMAES192),
+											string(network.IpsecEncryptionGCMAES256),
+											string(network.IpsecEncryptionNone),
+										}, false),
+									},
+									"ipsec_integrity_algorithm": {
```

Comment:
```suggestion
									"integrity_algorithm": {
```

Reply:
Instead, I think we shall rename the `ipsec_policy` block to `policy` since it actually define both ipsec and ike policy.

Reply:
see above comment

Reply:
Make sense! Will do the change suggested originally.